WalletWasabi.Fluent.Generators/UIContextGenerator.cs

```diff
@@ -0,0 +1,313 @@
+
+		var newSyntaxTrees =
+			ctors.Select(x => x.SyntaxTree)
+				 .Where(x => !compilation.ContainsSyntaxTree(x))
+				 .ToList();
+
+		compilation = compilation.AddSyntaxTrees(newSyntaxTrees);
+
+		foreach (var ctor in ctors)
+		{
+			var semanticModel = compilation.GetSemanticModel(ctor.SyntaxTree);
+
+			if (ctor.Parent is not ClassDeclarationSyntax cls)
+			{
+				continue;
+			}
+
+			if (!cls.IsRoutableViewModel(semanticModel))
+			{
+				continue;
+			}
+
+			var viewModelTypeInfo =
+				semanticModel.GetDeclaredSymbol(ctor.Parent as ClassDeclarationSyntax);
+
+			var className = cls.Identifier.ValueText;
+
+			var ctorNamespaces =
+					ctor.ParameterList.Parameters
+									  .Where(p => p.Type is not null)
+									  .Select(p => semanticModel.GetTypeInfo(p.Type!))
+									  .Where(t => t.Type is not null)
+									  .SelectMany(t => t.Type.GetNamespaces())
+									  .ToList();
+
+			var uiContextParam =
+				ctor.ParameterList
+					.Parameters
+					.FirstOrDefault(x => x.Type.IsUiContextType(semanticModel));
+
+			var methodParams = ctor.ParameterList;
+
+			if (uiContextParam != null)
+			{
+				methodParams = SyntaxFactory.ParameterList(methodParams.Parameters.Remove(uiContextParam));
+			}
+
+			var navigationMetadata =
+				viewModelTypeInfo.GetAttributes()
+						.FirstOrDefault(x => x.AttributeClass?.ToDisplayString() == NavigationMetaDataGenerator.NavigationMetaDataAttributeDisplayString);
+
+			var defaultNavigationTarget = "DialogScreen";
+
+			if (navigationMetadata != null)
+			{
+				var navigationArgument =
+					navigationMetadata.NamedArguments
+									  .FirstOrDefault(x => x.Key == "NavigationTarget");
+
+				if (navigationArgument.Value.Type is INamedTypeSymbol navigationTargetEnum)
+				{
+					var enumValue =
+						navigationTargetEnum.GetMembers()
+											.OfType<IFieldSymbol>()
+											.FirstOrDefault(x => x.ConstantValue?.Equals(navigationArgument.Value.Value) ?? false);
+
+					if (enumValue != null)
+					{
+						defaultNavigationTarget = enumValue.Name;
+					}
+				}
+			}
+
+			var additionalMethodParams =
+				$"NavigationTarget navigationTarget = NavigationTarget.{defaultNavigationTarget}, NavigationMode navigationMode = NavigationMode.Normal";
+
+			methodParams = methodParams.AddParameters(SyntaxFactory.ParseParameterList(additionalMethodParams).Parameters.ToArray());
+
+			var ctorArgs =
+				SyntaxFactory.ArgumentList(
+					SyntaxFactory.SeparatedList(
+					ctor.ParameterList
+						.Parameters
+						.Select(x => x.Type.IsUiContextType(semanticModel) ? "UiContext" : x.Identifier.ValueText) // replace uiContext argument for UiContext property reference
+						.Select(x => SyntaxFactory.ParseExpression(x))
+						.Select(SyntaxFactory.Argument),
+					ctor.ParameterList
+						.Parameters
+						.Skip(1)
+						.Select(x => SyntaxFactory.Token(SyntaxKind.CommaToken))));
+
+			namespaces.Add(viewModelTypeInfo.ContainingNamespace.ToDisplayString());
```

Comment:
`viewModelTypeInfo` is nullable, if it indeed can be null shouldn't it be checked?